api/envoy/extensions/filters/http/cdn_loop/v3alpha/cdn_loop.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.cdn_loop.v3alpha;
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.cdn_loop.v3alpha";
+option java_outer_classname = "CdnLoopProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: HTTP CDN-Loop Filter]
+// [#extension: envoy.filters.http.cdn_loop]
+
+// CDN-Loop Header filter config. See RFC 8586 for information about
+// the CDN-Loop header and its uses.
+message CdnLoopConfig {
+  // The CDN identifier to use for loop checks.
+  //
+  // RFC 8586 calls this the cdn-id. The cdn-id can either be a
+  // pseudonym or hostname the CDN is in control of.
+  //
+  // cdn_id must not be empty.
+  string cdn_id = 1 [(validate.rules).string = {min_bytes: 1}];
+
+  // The maximum allowed count of cdn_id in the downstream CDN-Loop
+  // request header.
+  //
+  // The default of 0 means a request can transit the CdnLoopFilter
```

Comment:
I suggest using `google.protobuf.UInt32Value` and saying that "not set" is once. Otherwise, it's confusing, since we have 0 and 1 meaning 1, and folks might thing you are doing -1 offset.

Reply:
0 and 1 have different meanings; the filter is doing a -1 offset since it's counting the number of times the incoming request has the CDN set.  If there's no header or the sought CDN value isn't there, the count is 0.  We then add the CDN value and send the request on its merry way.

If `max_allowed_occurrences` is 1, then an incoming request with no matching CDN value or an incoming request with 1 CDN value is allowed.  The request can transit the CDN protected by this filter twice.

I'm happy to rename the field if there's a better name (maybe `max_downstream_occurrences`?) or to beef up the documentation with examples.  What do you think?

Reply:
I beefed up the documentation.

Reply:
Got it, much clearer, thanks!